spring-shell2-shell1-adapter/src/main/java/org/springframework/shell2/legacy/LegacyParameterResolver.java

```diff
@@ -113,17 +112,67 @@ public ParameterDescription describe(MethodParameter parameter) {
 
 	@Override
 	public List<CompletionProposal> complete(MethodParameter parameter, CompletionContext context) {
-		return null;
+		// DevNote: in Legacy the user is either completing an argument (-- prefix) or the value of the argument (e.g. no arity to consider).
+		// Other cases to be considered include default keys and default values
+		
+		CliOption option = parameter.getParameterAnnotation(CliOption.class);
+		List<String> keys = Arrays.asList(option.key());
+		
+		int currentWordIndex = context.getWordIndex();		
+		
+		// 1) has a key for this parameter already been defined?
+		boolean keyAlreadyDefined = context.getWords().stream()
```

Comment:
I've been giving this some thought and we might be repeating ourselves with this type of check. 

As in, the parameters themselves are double checking if they've already been defined or resolved. 

In the case of the `StandardParameterResolver` we are attempting to resolve before checking for completions, and although I think that's the correct approach, I wonder if this responsibility should be handled at a higher level/layer?

Should we have a `MethodArgumentResolver` that has this type of responsibilities? This could also tie into issue #48 where the `MethodArgumentResolver` could populate the context with additional details as it resolves Parameters.

Currently, we go from `JLineShell` -> `ParameterResolver` directly. My proposal would be to have `JLineShell` -> `MethodArgumentResolver` -> `ParameterResolver` (the name of `MethodArgumentResolver` to be discussed :) )

This way the `MethodArgumentResolver` can keep track of parameters that have already been resolved vs those that haven't and only request completions from unresolved parameters.

This seemed large enough to deserve it's own PR, so didn't want to include changes here to keep things separate :)

Let me know your thoughts on this.